### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by mathbunnyru**

### CODE REVIEW

**Comment by mathbunnyru** [comment]
```
#11 465.8 Found conflicts! Looking for incompatible packages.
#11 465.8 This can take several minutes.  Press CTRL-C to abort.
#11 465.8 failed
#11 465.8
#11 465.8 UnsatisfiableError: The following specifications were found
#11 465.8 to be incompatible with the existing python installation in your environment:
#11 465.8
#11 465.8 Specifications:
#11 465.8
#11 465.8   - beautifulsoup4=4.9 -> python[version='3.6.*|3.7.*|3.8.*|>=2.7,<2.8.0a0']
#11 465.8   - dill=0.3 -> python[version='2.7.*|3.6.*|3.8.*|3.7.*']
#11 465.8   - patsy=0.5 -> python[version='2.7.*|3.5.*|3.6.*']
#11 465.8   - vincent=0.4 -> python[version='2.7.*|3.4.*|3.5.*|3.6.*']
#11 465.8
#11 465.8 Your python: python=3.9
```

:(

File: docs/using/selecting.md
Comment (mathbunnyru) [inline docs/using/selecting.md:99]: Vincent is deprecated from 2016-06-18.
It is recommended to install altair, so I replace the library.
https://github.com/wrobstory/vincent#2016-06-18-update

Altair is a nice tool, I've been using it for quite a while.

File: all-spark-notebook/Dockerfile
Comment (mathbunnyru) [inline all-spark-notebook/Dockerfile:25]: This takes too much time with conda.
I replace the parts which were taking too long with conda on my machine.

Let's try mamba, it seems to work great.

**Comment by mathbunnyru** [comment]
@romainx please review, this change is not as simple as usual one.

File: all-spark-notebook/Dockerfile
Comment (consideRatio) [reply all-spark-notebook/Dockerfile:25]: I'm very supportive of this transition, it has worked out great and made a significant difference in performance.

**Review by consideRatio — APPROVED** [review]
LGTM!

**Comment by mathbunnyru** [comment]
Actually, the build time is reduced significantly.
Last master build: https://github.com/jupyter/docker-stacks/runs/2645915410
`35m 48s`

This PR build:
https://github.com/jupyter/docker-stacks/pull/1319/checks?check_run_id=2645918540
`19m 24s`

File: all-spark-notebook/Dockerfile
Comment (romainx) [reply all-spark-notebook/Dockerfile:25]: On this point the last time we tried to switch to mamba we noticed differences on the size of images. It seemed that something was missing (in fact we had no clear explanation at the time) in R images. See [here](https://github.com/jupyter/docker-stacks/pull/1213#issuecomment-761517281) for the summary. I think we should check if we have the same behavior here before performing this change.

File: docs/using/selecting.md
Comment (romainx) [inline docs/using/selecting.md:98]: @mathbunnyru very good suggestion. 👍

**Review by romainx — COMMENTED** [review]
Hello,

Sorry for not being too available at the moment 😕.
I had a look to this PR and I think it's all good be we should have a look at the usage of Mamba for R images, since last time we noticed something weird (see my specific comment). At least we should be aware of that to be able to fix it after a merge.

Thanks for all this work 👍

File: docs/using/selecting.md
Comment (mathbunnyru) [reply docs/using/selecting.md:98]: Thanks :)

**Comment by mathbunnyru** [comment]
As @romainx asked, I will take a close look, what actually changes, when using mamba for r packages.

**Comment by mathbunnyru** [comment]
GitHub is being weird here.
I have changed this PR quite significantly after @consideRatio's review, but it still tells me that this PR is approved.

**Comment by manics** [comment]
@mathbunnyru there should be a repository setting to dismiss stale reviews, e.g.
image
If it's already set then that sounds like a GitHub bug....

**Comment by mathbunnyru** [comment]
Thanks, @manics. I think this setting is not set.
I asked to add this setting (I don't have such permissions).

**Comment by parente** [comment]
@mathbunnyru Review dismissal upon new commits is now checked. I had it unchecked when I was the sole maintainer and reviews never happened.

**Comment by mathbunnyru** [comment]
@parente and now this is weird, because the review by @consideRatio still counts as a successful one, even when I changed a new file (for a test).

**Comment by parente** [comment]
@mathbunnyru That is weird. Can you try again now? I toggled the setting off and on again.

**Comment by parente** [comment]
Proof it's set as of the time of this comment, in case something is reverting it.

image

**Comment by mathbunnyru** [comment]
@parente it's working now!

But now I'm not able to merge my PRs without a review (even this one).
That's not something I wanted 😆 
And actually, that's something dangerous - for example, if some bad commit is merged, we need to revert it and it's not an easy process when one review is required.
There are sometimes easy changes, which do not require maintainers' attention.

Is it possible to set the value of "Required approving reviews" to zero?
And this way keep the setting about dismissing reviews.

If not, then please disable this setting and let's keep everything as it was :(

Sorry for annoying you too much.

**Review by consideRatio — APPROVED** [review]
Still LGTM!

**Comment by parente** [comment]
@mathbunnyru The lowest number of reviews I can pick from the drop down is 1. 

Having cake and eating cake are mutually exclusive here. 😂 

I’ll set it back the way it was after finishing this comment.

And no need for apologies. Thank you for taking over the maintenance and improvement efforts here!

**Comment by mathbunnyru** [comment]
After merging mamba in a separate PR, this is an easy update. I'll merge.

**Comment by consideRatio** [comment]
Great work @mathbunnyru!! ♥️🥳

**Comment by mathbunnyru** [comment]
> Great work @mathbunnyru!! ♥️🥳

Thanks 😊😊

**Pull request merged**
